lightshow/postprocess/compare_utils.py

```diff
@@ -0,0 +1,258 @@
+# This is a simple tool to compare to plots
+# Chuntian Cao developed based on Fanchen Meng's 2022 code
+
+
+import numpy as np
+from scipy import interpolate
+from scipy.stats import pearsonr, spearmanr
+
+
+def compare_between_spectra(
+    spectrum1, spectrum2, erange=35, accuracy=0.01, method="coss"
+):
+    """Atomatic align the spectra and calculate the spearman coefficient
+    Comparison was done for epsilon instead of cross-section
+
+    Parameters
+    ----------
+    spectrum1 and spectrum2 : two-column arrays of energy vs. intensity XAS
+    method : 'pearson', 'spearman', or 'coss' (cosine similarity).
+        Empirically 'coss' works well
+
+    Return
+    ------
+    pearson, spearman, cossine correlation : real
+    shift : real; relative shift between the two spectra, sign is meaningful
+    """
+
+    start1, end1 = truncate_spectrum(spectrum1, erange)
+    plot1 = np.column_stack(
+        (
+            spectrum1[start1:end1, 0] - spectrum1[start1][0],
+            spectrum1[start1:end1, 1],
+        )
+    )
+    start2, end2 = truncate_spectrum(spectrum2, erange)
+    plot2 = np.column_stack(
+        (
+            spectrum2[start2:end2, 0] - spectrum2[start2][0],
+            spectrum2[start2:end2, 1],
+        )
+    )
+
+    shift_prior = spectrum1[start1, 0] - spectrum2[start2, 0]
+    _, shift = maxCorr(plot1, plot2, step=accuracy, method=method)
+    pears, spear, coss = spectraCorr(
+        plot1, plot2, omega=shift, verbose=True, method="all"
+    )
+    shift += shift_prior
+
+    return pears, spear, coss, shift
+
+
+def truncate_spectrum(spectrum, erange=35, threshold=0.02):
+    """
+    Parameters
+    ----------
+    spectrum: column stacked spectrum, energy vs. intensity
+    erange: truncation range
+    threshold: start truncation at threshold * maximum intensity
+
+    Return
+    ------
+    start, end : int; indices of truncated spectrum in the input spectrum
+    """
+    x = spectrum[:, 0]
+    y = spectrum[:, 1] / np.max(spectrum[:, 1])
+
+    logic = y > threshold
+    seq = x == x[logic][0]
+    start = seq.argmax()
+
+    logic = x < x[start] + erange
+    seq = x == x[logic][-1]
+    end = seq.argmax()
+
+    return start, end
+
+
+def CosSimilar(v1, v2):
```

Comment:
Minor ask here, can we make the function styles consistent? I.e., this should be `cos_similar`.